### PR TITLE
fixes #21823 - adds foreman user agent to RSS loading

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: https://tools.ietf.org/html/rfc7231#section-5.5.3 suggests the common way is to separate the product and version with a slash. I'm also not sure about the RSS notifications part of it but if you do, convention is to put it in braces: ```Foreman/#{SETTINGS[:version]} (RSS notifications)```

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: Please do verify SSL or at least provide an option to disable it (but default to verification). We should assume correct SSL stacks and not trust incorrect ones.

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: If you use this instead of the ```Net::HTTP.start(uri.host, uri.port) do |http|``` form, does it correctly close the connection after this function or will it wait for the garbage collection?

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: An option to opt-out of changing this should be considered.

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: The documentation at https://ruby-doc.org/stdlib-2.4.2/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-HTTPS suggests this is no longer needed if you just pass in the uri rather than the ```uri.request_uri```.

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: https://ruby-doc.org/stdlib-2.4.2/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-HTTPS states
> In previous versions of Ruby you would need to require 'net/https' to use HTTPS. This is no longer true.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ohadlevy) [reply app/services/ui_notifications/rss_notifications_checker.rb]: it should close the connection once you call the http.request.body

**Comment by ohadlevy** [comment]
@ekohl updated.

**Comment by ohadlevy** [comment]
@dLobatog mind having a look? thanks :)

**Review by ekohl — APPROVED** [review]
I'll leave it to the core team to review further.

**Comment by ohadlevy** [comment]
in the web server logs, the following will be observed:
```
"GET /feed.xml HTTP/1.1" 200 83237 "-" "Foreman/1.17.0-develop (RSS notifications)"
```

**Review by boaz0 — APPROVED** [review]
LGTM 🏆

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ares) [reply app/services/ui_notifications/rss_notifications_checker.rb]: I don't think `http.request.body` closes the connection, but it looks like HTTP#request [runs](http://ruby-doc.org/stdlib-1.9.3/libdoc/net/http/rdoc/Net/HTTP.html) the request in a start using the block. Though I agree, using start with block would be more readable and ruby-like.

**Review by ares — CHANGES_REQUESTED** [review]
Tested, works well. @ohadlevy while the code really closes the HTTP connection, would you mind changing it to `start` with block so it's clear from first read? Then it's good to get in.

**Comment by ares** [comment]
Also it needs rebase now.

**Comment by ohadlevy** [comment]
> I don't think http.request.body closes the connection, but it looks like HTTP#request runs the request in a start using the block. Though I agree, using start with block would be more readable and ruby-like.

@ares what would you suggest I change? I didnt think do a block and return inside the block would be better?

(rebased regardless so tests are executed).

**Comment by ares** [comment]
> @ares what would you suggest I change? I didnt think do a block and return inside the block would be better?

not sure why you'd need to return, see https://github.com/ohadlevy/foreman/pull/10, works just fine, the docs says `#start` [returns the return value of a block](https://docs.ruby-lang.org/en/2.0.0/Net/HTTP.html#method-i-start)

File: app/services/ui_notifications/rss_notifications_checker.rb
Comment (ekohl) [inline app/services/ui_notifications/rss_notifications_checker.rb]: Should there be any error checking for incorrect HTTP status codes? It's possible the host is unreachable, misconfigured or a proxy is refusing it.

**Comment by ohadlevy** [comment]
@ares would you squash on merge?

**Comment by ares** [comment]
I'll be happy to squash, I sent another PR at https://github.com/ohadlevy/foreman/pull/11 to address @ekohl's comment. @ohadlevy could you please merge that one as well? I guess then it's ready to be merged.

**Review by ekohl — APPROVED** [review]

**Review by ares — APPROVED** [review]

**Review by ares — CHANGES_REQUESTED** [review]
the change has broken tests, I'll send another addition

**Comment by ares** [comment]
tests fixed at https://github.com/ohadlevy/foreman/pull/12, the new code needed more stubbing

**Comment by ares** [comment]
now Foreman tests failed on HostgroupJSTest on a single combination, retriggering [test foreman], once it's green I'll squash and merge

**Review by ares — APPROVED** [review]

**Comment by ohadlevy** [comment]
foreman tests are :green_apple: - katello failures seem unrelated?

**Comment by ares** [comment]
Thanks @ohadlevy, merging. @GregSutcliffe @orrabin you might be interested in this.

**Pull request merged**
